php/proxy.php

```diff
@@ -3,150 +3,192 @@
  * Acrolinx Proxy
  * Description - Creates proxy for Acrolinx.
  * Author - Acrolinx
- * Version - 1.0
+ * Version - 1.1
  */
- 
+
 class Proxy {
+
     private $settings;
     private $targetPath;
     private $username;
     private $password;
-    private $domainURL;	
-	
+    private $domainURL;
+    private $host;
+    private $url;
+
     private function disable_gzip() {
         @ini_set('zlib.output_compression', 'Off');
-        @ini_set('output_handler', '');	
+        @ini_set('output_handler', '');
     }
-    
-    
+
     private function getSettings() {
         $acrolinxSettings = array();
-
         $acrolinxSettings['isSSO'] = 1; //TODO: Set if SSO should be used, otherwise it will be just a reverse proxy.
         $acrolinxSettings['url'] = ""; //TODO: Set Acrolinx server url here.
         $acrolinxSettings['genericPassword'] = "secret"; //TODO: Set secret token here.
 
         if($acrolinxSettings['isSSO'] == 1) {
-
             $acrolinxSettings['password'] = $acrolinxSettings['genericPassword'];
             $acrolinxSettings['username'] = get_current_user(); //TODO: Implement get_current_user, which retrieves the username from the current session.
         }
         return $acrolinxSettings;
     }
-    
+
     private function getPosition($requestURI, $part) {
         return strpos($requestURI, $part);
     }
-    
+
     private function getActualURI($requestURI, $pos, $part) {
         return substr_replace($requestURI, '', 0, ($pos+strlen($part)));
     }
-    
+
+    private function emulate_getallheaders() {
+        foreach ($_SERVER as $name => $value) {
+            if (substr($name, 0, 5) == 'HTTP_') {
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
```

Comment:
What's this for? What will happen to the request? Is that relevant for the Acrolinx Server later on? Why first to lower and then the sting might look funny like "HTTP-Test-Foo-Bar". Is that wanted?

Reply:
A1: getallheaders() doesn't work on **Nginx** server. This is useful if server used in Nginx instead of Apache. While installing Drupal plugin on **Acquia Cloud** this issue was seen. It's an off the shelf function taken from PHP manual comment. (http://php.net/manual/en/function.getallheaders.php)
A2/A3: This is a  generic function. It should not have any impact on the request reaching Acrolinx server.
A4: The string won't look like HTTP-Test-Foo-Bar rather it would be Test-Foo-Bar (substr($name, 5))

Solution: I'll make this function a fallback. It native getallheaders doesn't exist use this